app/controllers/omniauth_controller.rb

```diff
@@ -0,0 +1,65 @@
+class OmniauthController < Devise::OmniauthCallbacksController
+  skip_before_action :verify_authenticity_token, only: [:tra_openid_connect]
+  before_action :remove_from_get_an_identity_pilot, only: :tra_openid_connect, if: :tra_response_excludes_trn?
```

Comment:
Without a TRN the user can't finish their application, to resolve this we redirect them out of the pilot.